src/foam/nanos/auth/AuthenticatedAgentJunctionDAO.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.dao.Sink;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.util.SafetyUtil;
+
+import static foam.mlang.MLang.EQ;
+import static foam.mlang.MLang.OR;
+import static foam.mlang.MLang.AND;
+
+/**
+ * Authenticated AgentJunctionDAO
+ * Restrict users from creating a junction object where they are the source user (Agent)
+ * Restrict users from finding and selecting junction objects they are not referenced to. (Agent or Entity)
+ * Restrict users from removing junction objects where they are the source user (Agent)
```

Comment:
Shouldn't this be from removing junction objects where they aren't the target user? This sounds like I could remove other people's junctions.